src/NServiceBus.Core/Routing/MessageDrivenSubscriptions/MessageType.cs

```diff
@@ -96,7 +96,7 @@ public bool Equals(MessageType other)
             {
                 return true;
             }
-            return Equals(other.TypeName, TypeName) && other.Version.Major == Version.Major;
+            return Equals(other.TypeName, TypeName);
```

Comment:
while you're right, I don't see an issue with this being a breaking change? I think we never considered this a non-breaking change?